File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        
        vector<vector<int>>ans;
        if(!root)
            return ans;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty())
        {
            int num_nodes = q.size();
            vector<int>v1;
            for(int i = 0; i < num_nodes; i++)
            {
                TreeNode* n1 = q.front();
                v1.push_back(n1->val);
                q.pop();
                if(n1->left)
                    q.push(n1->left);
                if(n1->right)
                    q.push(n1->right);
            }
            ans.push_back(v1);
        }
        return ans;
    }
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {


        List<List<Integer>>list1 = new ArrayList<>();
        if(root == null){
            return list1;
        }
        Queue<TreeNode>l1 = new ArrayDeque<>();
        l1.add(root);
        while(!l1.isEmpty())
        {
            int size_l = l1.size();
            List<Integer>list2 = new ArrayList<>();
            for(int i = 0 ; i < size_l ; i++)
            {
                TreeNode node = l1.poll();
                list2.add(node.val);
                if(node.left!=null)
                    l1.add(node.left);
                if(node.right!=null)
                    l1.add(node.right);
            }
            list1.add(list2);
        }
        return list1;
    }
}